### PR TITLE
Various translation fixes

### PR DESCRIPTION
**Pull request by GeorgeK1ng**
1) Updated Czech translation
2) Fixed coalesting text space in Pyramid, same as last fix for Shrine
3) Fixed ImpCache definition name typo
4) Updated Russian translation with corrected ImpCache instead of InpCache

### CODE REVIEW

**Comment by dydzio0614** [comment]
unless @IvanSavenko recommends otherwise I would revert imp cache typo fix for now as it has severe consequences

**Comment by GeorgeK1ng** [comment]
> unless @IvanSavenko recommends otherwise I would revert imp cache typo fix for now as it has severe consequences

Let's see what Ivan found then, I have uploaded fixes for all dependends mods aswell.

**Comment by GeorgeK1ng** [comment]
I have searched across both VCMI and Mods repositories and it should be fine, when applied in main bank definition, then in dependend mods. InpCache is not hardcoded anywhere from my search in GitHub, just 3 mods depency.

**Comment by misiokles** [comment]
There is also one bug in fortress faction. Special1 and special2 buildings add +2 defense to garrosoned heroes, but one should add +2 defense, second +2 attack.

**Comment by Olegmods** [comment]
> There is also one bug in fortress faction. Special1 and special2 buildings add +2 defense to garrosoned heroes, but one should add +2 defense, second +2 attack.

Fixed in #4842

File: config/objects/creatureBanks.json [hunk 1]
```diff
@@ -261,7 +261,7 @@
 					}
 				]
 			},
-			"inpCache" :  {
+			"impCache" :  {
```
Comment (IvanSavenko) [inline config/objects/creatureBanks.json:264]:
```suggestion
			"impCache" :  {
				"compatibilityIdentifiers" : [ "inpCache" ],
```

**Comment by IvanSavenko** [comment]
Should be safe I think.
The only thing that may actually break would be maps with, ahem, inpCache strings in them. Can be fixed via compatibilityIdentifiers field

File: config/objects/creatureBanks.json (hunk 1 above)
Comment (Laserlicht) [reply config/objects/creatureBanks.json:264]: But would be better anyway to also fix it in any translation mod?

Comment (GeorgeK1ng) [reply config/objects/creatureBanks.json:264]: It exist as InpCache only in Russian translation.

Comment (Laserlicht) [reply config/objects/creatureBanks.json:264]: In most cases these strings are in the translation mod.

Examples:
- https://github.com/vcmi-mods/german-translation/blob/vcmi-1.6/content/config/vcmi-german/game.json
- https://github.com/vcmi-mods/french-translation/blob/vcmi-1.6/french-translation/content/config/vcmi-french/game.json

Comment (GeorgeK1ng) [reply config/objects/creatureBanks.json:264]: There is something I don't understand. That's why I told it's not in translations. Will check it manually instead.
image

Comment (misiokles) [reply config/objects/creatureBanks.json:264]: github search options can't read string from files, only in filenames

Comment (IvanSavenko) [reply config/objects/creatureBanks.json:264]: Looks like it

> Beyond that, we also don't include all files in the search index:
    Vendored and generated code is excluded
    Empty files and files over 350 kiB are excluded
    Only UTF-8 encoded files are indexed
    Very large repositories may not be indexed

Comment (Laserlicht) [reply config/objects/creatureBanks.json:264]: I think `mega-pack-rus` and `h3-for-vcmi-englisation` are missing (for historical reasons there isn't `translation` in mod name)

Comment (GeorgeK1ng) [reply config/objects/creatureBanks.json:264]: Created PRs for them aswell now. Hope thats all

Comment (misiokles) [reply config/objects/creatureBanks.json:264]: I polish there's as well inpCache

Comment (GeorgeK1ng) [reply config/objects/creatureBanks.json:264]: > I polish there's as well inpCache. I guess all translations have exported inpCache string, not impCache

Polish PR is already done too

**Pull request merged**